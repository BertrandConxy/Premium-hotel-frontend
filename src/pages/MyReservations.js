import React from 'react';
import phfrom from '../images/ph.png';
import NavBar from '../components/NavBar';

function MyReservation() {
  const reservations = [1, 2, 3, 4, 5];
  return (
    <div className="home">
      <div className="nav">
        <img src={phfrom} className="ph-logo" alt="" />
        <NavBar />
      </div>
      <div className="main">
        <h1>My reservations</h1>
        <h2>
          You are able to cancel the reservation before 24 hours of the
          reservation date
        </h2>
        <div className="reservation">
          {reservations.map((reservation) => (
            <NavLink to="/reserv" exact="true" key={reservation} />
          ))}
        </div>
      </div>
    </div>
  );
}

export default MyReservation;
