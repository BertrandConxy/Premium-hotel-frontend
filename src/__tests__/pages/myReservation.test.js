import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import { screen, render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import MyReservations from "../../pages/MyReservations";
import store from "../../redux/configureStore";

const ReservationsProvider = () => (
  <Provider store={store}>
    <Router>
      <MyReservations />
    </Router>
  </Provider>
);

describe("render component", () => {
  it("renders the rooms list form", () => {
    render(<ReservationsProvider />);
    const heading = screen.getByText("My reservations");
    expect(heading).toBeInTheDocument();
  });
});

describe("Snapshot test", () => {
  it("looks the same as the snapshot", () => {
    const component = renderer.create(<ReservationsProvider />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
