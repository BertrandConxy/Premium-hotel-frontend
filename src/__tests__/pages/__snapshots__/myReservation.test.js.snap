// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test looks the same as the snapshot 1`] = `
<div
  className="home"
>
  <button
    className="p-2 vis btn w-25"
    type="button"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-bars "
      data-icon="bars"
      data-prefix="fas"
      focusable="false"
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 96C0 78.33 14.33 64 32 64H416C433.7 64 448 78.33 448 96C448 113.7 433.7 128 416 128H32C14.33 128 0 113.7 0 96zM0 256C0 238.3 14.33 224 32 224H416C433.7 224 448 238.3 448 256C448 273.7 433.7 288 416 288H32C14.33 288 0 273.7 0 256zM416 448H32C14.33 448 0 433.7 0 416C0 398.3 14.33 384 32 384H416C433.7 384 448 398.3 448 416C448 433.7 433.7 448 416 448z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </button>
  <div
    className="nav pt-10"
  >
    <img
      alt="Premium Hotel Logo"
      className="lunar-logo"
      src="lunar.png"
    />
    <section>
      <nav>
        <ul>
          <li
            className="nav-link the-nav pt-0 pb-0 float-left text-black"
          >
            <a
              activeclassname="active"
              aria-current="page"
              className="active"
              exact="true"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li
            className="nav-link the-nav pt-0 pb-0 float-left text-black"
          >
            <a
              activeclassname="active"
              className=""
              exact="true"
              href="/reservations"
              onClick={[Function]}
            >
              My Reservations
            </a>
          </li>
          <li
            className="nav-link the-nav pt-0 pb-0 float-left text-black"
          >
            <a
              activeclassname="active"
              className=""
              exact="true"
              href="/create_room"
              onClick={[Function]}
            >
              Create Room
            </a>
          </li>
          <li
            className="nav-link the-nav pt-0 pb-0 float-left text-black"
          >
            <a
              activeclassname="active"
              className=""
              exact="true"
              href="/rooms"
              onClick={[Function]}
            >
              Delete Room
            </a>
          </li>
          <li
            className="nav-link the-nav pt-0 pb-0 float-left text-black"
          >
            <a
              activeclassname="active"
              className=""
              exact="true"
              href="/add_reservation"
              onClick={[Function]}
            >
              Reserve Room
            </a>
          </li>
        </ul>
      </nav>
    </section>
  </div>
  <div
    className="main"
  >
    <h1
      className="fs-1 fw-bold text-center"
    >
      My reservations
    </h1>
    <h2
      className="no-reservation lh-lg mt-5 fs-4"
    >
      Oh Oh!
      <br />
      You do not have a reservation yet
      <br />
      Kindly reserve a room on the 
      <a
        className="text-danger"
        href="/"
      >
        HOME
      </a>
       page
    </h2>
    ;
    <div
      className="reservations"
    />
  </div>
</div>
`;
